app/models/post.rb

```diff
@@ -1,15 +1,18 @@
 class Post < ApplicationRecord
-  belongs_to :users
   has_many :comments
   has_many :likes
+  belongs_to :users, foreign_key: 'user_id', class_name: 'User'
+  after_create :update_posts_counter
 
-  # A method that updates the posts counter for a user.
-  def update_user_posts_counter
-    user.update(posts_counter: user.posts.count)
-  end
+  validates :title, presence: true, length: { in: 3..250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
-  # A method which returns the 5 most recent comments for a given post.
-  def recent_comments
+  def return_recent_comments
     comments.order(created_at: :desc).limit(5)
   end
+
+  def update_posts_counter
+    users.increment!(:posts_counter)
+  end
```

Comment:
[OPTIONAL] These two methods (return_recent_comments, and update_posts_counter) should have a corresponding unit test.